src/init.c

```diff
@@ -753,6 +753,8 @@ static NOINLINE void _finish_julia_init(JL_IMAGE_SEARCH rel, jl_ptls_t ptls, jl_
         JL_GC_POP();
     }
 
+    jl_gc_init_finalizer_rng_state();
```

Comment:
Do we have a guarantee to have randomness by now? And not expect finalizers to attempt to need randomness before now?

Reply:
sounds best to me